apps/wing/src/commands/test/test.ts

```diff
@@ -357,9 +364,98 @@ async function runTestsWithRetry(
   return results;
 }
 
+function inferSeverityOfEvent(trace: std.Trace): "error" | "warn" | "info" | "debug" | "verbose" {
+  if (trace.data.status === "failure") {
+    return "error";
+  }
+  if (trace.type === TraceType.LOG) {
+    return "info";
+  }
+  if (trace.type === TraceType.RESOURCE) {
+    return "debug";
+  }
+  if (trace.type === TraceType.SIMULATOR) {
+    return "verbose";
+  }
+  return "verbose";
+}
+
+const SEVERITY_STRING = {
+  error: "ERROR  ",
+  warn: "WARNING",
+  info: "INFO   ",
+  debug: "DEBUG  ",
+  verbose: "VERBOSE",
+};
+
+const LOG_STREAM_COLORS = {
+  error: chalk.red,
+  warn: chalk.yellow,
+  info: chalk.green,
+  debug: chalk.blue,
+  verbose: chalk.gray,
+};
+
 async function testSimulator(synthDir: string, options: TestOptions) {
   const s = new simulator.Simulator({ simfile: synthDir });
   const { clean, testFilter, retry } = options;
+
+  let outputStream: SpinnerStream | undefined;
+  if (options.stream) {
+    outputStream = new SpinnerStream(process.stdout, "Running tests...");
+
+    const testMappings = extractTestMappings(s.listResources());
+
+    s.onTrace({
+      callback: (event) => {
+        // Filter out simulator and resource events if DEBUG isn't set
+        if (
+          (event.type === TraceType.SIMULATOR || event.type === TraceType.RESOURCE) &&
+          !process.env.DEBUG
+        ) {
+          return;
+        }
+
+        // Filter out simulator events if DEBUG=verbose isn't set
+        if (event.type === TraceType.SIMULATOR && process.env.DEBUG !== "verbose") {
+          return;
+        }
+
+        const env = extractTestEnvFromPath(event.sourcePath);
+        if (env === undefined) {
+          // This event is not associated with any test environment, so skip it.
+          return;
+        }
+        const testName = testMappings[env];
+        if (testName === undefined) {
+          // This event is not associated with any test environment, so skip it.
+          return;
+        }
+        if (testFilter && !testName.includes(testFilter)) {
+          // This test does not match the filter, so skip it.
+          return;
+        }
+
+        const pathSuffix = event.sourcePath.split("/").slice(2).join("/");
+        const severity = inferSeverityOfEvent(event);
+        const date = new Date(event.timestamp);
+        const hours = date.getHours().toString().padStart(2, "0");
+        const minutes = date.getMinutes().toString().padStart(2, "0");
+        const seconds = date.getSeconds().toString().padStart(2, "0");
+        const milliseconds = date.getMilliseconds().toString().padStart(3, "0");
+        const timestamp = `${hours}:${minutes}:${seconds}.${milliseconds}`;
+        let msg = "";
+        msg += chalk.gray(`[${timestamp}]`);
```

Comment:
Try this one to see if it's any good. I feel like the timestamp is a little too prominent.

```suggestion
        msg += chalk.dim(`[${timestamp}]`);
```

Reply:
Done. dim does look a little darker on my screen